ctapipe/instrument/camera.py

```diff
@@ -308,32 +321,74 @@ def __str__(self):
 
     @lazyproperty
     def neighbors(self):
-        """" only calculate neighbors when needed or if not already
-        calculated"""
-
-        # return pre-calculated ones (e.g. those that were passed in during
-        # the object construction) if they exist
-        if self._precalculated_neighbors is not None:
-            return self._precalculated_neighbors
-
-        # otherwise compute the neighbors from the pixel list
-        dist = _get_min_pixel_seperation(self.pix_x, self.pix_y)
-
-        neighbors = _find_neighbor_pixels(
-            self.pix_x.value,
-            self.pix_y.value,
-            rad=1.4 * dist.value
-        )
-
-        return neighbors
+        '''A list of the neighbors pixel_ids for each pixel'''
+        return [np.where(r)[0].tolist() for r in self.neighbor_matrix]
 
     @lazyproperty
     def neighbor_matrix(self):
-        return _neighbor_list_to_matrix(self.neighbors)
+        return self.neighbor_matrix_sparse.A
 
     @lazyproperty
     def neighbor_matrix_sparse(self):
-        return csr_matrix(self.neighbor_matrix)
+        if self._neighbors is not None:
+            return self._neighbors
+        else:
+            return self.calc_pixel_neighbors(diagonal=False)
+
+    def calc_pixel_neighbors(self, diagonal=False):
+        '''
+        Calculate the neighbors of pixels using
+        a kdtree for nearest neighbor lookup.
+
+        Parameters
+        ----------
+        diagonal: bool
+            If rectangular geometry, also add diagonal neighbors
+        '''
+        neighbors = lil_matrix((self.n_pixels, self.n_pixels), dtype=bool)
+
+        if self.pix_type.startswith('hex'):
+            k = 7
+            radius = 1.4
```

Comment:
Ah sorry one more thing. I believe these 3 `radius` are somehow magic numbers. Maybe it would be good to shortly explain their choice in a comment. Maybe each `radius` needs and individual comment maybe not? 

I think you already commented on them in this discussion, but this will be lost after the PR is merged.

Reply:
done